RadixWallet/Core/SharedModels/Assets/OnLedgerEntity.swift

```diff
@@ -588,20 +588,32 @@ extension OnLedgerEntity.Account {
 	}
 }
 
-extension OnLedgerEntitiesClient.ResourceWithVaultAmount {
-	func poolRedemptionValue(for poolUnitsAmount: RETDecimal, poolUnitResource: OnLedgerEntity.Resource) -> String {
-		guard let poolUnitTotalSupply = poolUnitResource.totalSupply else {
-			loggerGlobal.error("Missing total supply for \(poolUnitResource.resourceAddress.address)")
+extension OnLedgerEntity.Resource {
+	func poolRedemptionValue(for amount: RETDecimal, poolUnitResource: OnLedgerEntitiesClient.ResourceWithVaultAmount) -> String {
+		guard let poolUnitTotalSupply = poolUnitResource.resource.totalSupply else {
+			loggerGlobal.error("Missing total supply for \(poolUnitResource.resource.resourceAddress.address)")
 			return L10n.Account.PoolUnits.noTotalSupply
 		}
 		guard poolUnitTotalSupply > 0 else {
-			loggerGlobal.error("Total supply is 0 for \(poolUnitResource.resourceAddress.address)")
+			loggerGlobal.error("Total supply is 0 for \(poolUnitResource.resource.resourceAddress.address)")
 			return L10n.Account.PoolUnits.noTotalSupply
 		}
-		let redemptionValue = poolUnitsAmount * (amount / poolUnitTotalSupply)
-		let decimalPlaces = resource.divisibility.map(UInt.init) ?? RETDecimal.maxDivisibility
+		let redemptionValue = poolUnitResource.amount * (amount / poolUnitTotalSupply)
+		let decimalPlaces = divisibility.map(UInt.init) ?? RETDecimal.maxDivisibility
 		let roundedRedemptionValue = redemptionValue.rounded(decimalPlaces: decimalPlaces)
 
 		return roundedRedemptionValue.formatted()
```

Comment:
I really don't like this, I'd rather do this work twice, saving clock cycles is not worth it!